CorinneTest/Factory.cs

```diff
@@ -0,0 +1,195 @@
+using System.Collections.Generic;
+
+namespace CorinneTest
+{
+
+
+    public class Factory 
+
+    {
+
+        static Random rnd = new Random();
+
+
+        public static List<Person> CreateInRange(int lowAge,
+                                                 int highAge,
+                                                 int? numOfPeople)
+        {
+            var list = new List<Person>();
+            int i = 0;
+            while (i < numOfPeople)
+            {
+                Person newPerson = null;
+                switch (rnd.Next(1, 9))
+                {
+                    case 1:
+                        newPerson = MakeWoman(lowAge, highAge);
+                        break;
+                    case 2:
+                        newPerson = MakeMan(lowAge, highAge);
+                        break;
+                    case 3:
+                        newPerson = MakeSomeone(lowAge, highAge);
+                        break;
+                    case 4:
+                        newPerson = MakeGirl(lowAge, highAge);
+                        break;
+                    case 5:
+                        newPerson = MakeBoy(lowAge, highAge);
+                        break;
+                    case 6:
+                        newPerson = MakeChild(lowAge, highAge);
+                        break;
+                    case 7:
+                        newPerson = MakeSeniorWoman(lowAge, highAge);
+                        break;
+                    case 8:
+                        newPerson = MakeSeniorMan(lowAge, highAge);
+                        break;
+                    default:
+                        newPerson = MakeSeniorPerson(lowAge, highAge);
+                        break;
+
+
+                }
+                list.Add(newPerson);
+                i++;
+
+            }
+            return list;
+        }
+        #region various genBirthDates
+        public static DateTime? GenRanAdultBirthDate(int lowAge, int highAge)
+        {
+            if (highAge < 18)
+                return null;
+            var start = new DateTime((2018 - highAge), 5, 29);
+            var minDate = new DateTime((2018 - lowAge), 5, 29);
+            int range = (minDate - start).Days;
+            return start.AddDays(rnd.Next(range));
+        }
+
+        public static DateTime? GenRanChildBirthDate(int lowAge, int highAge)
+        {
+            DateTime maxDate;
+            if (lowAge >= 18) { return null; }
+            if (highAge < 18) { maxDate = new DateTime((2018 - highAge), 5, 29); }
+            else { maxDate = new DateTime(2000, 5, 29); }
+            int range = (DateTime.Today - maxDate).Days;
+            return DateTime.Today.AddDays(rnd.Next(range));
+        }
+
+        public static DateTime? GenRanSeniorBirthDate(int lowAge, int highAge)
+        {
+            DateTime minDate;
+            if (highAge < 6) { return null; }
+            if (lowAge > 65) { minDate = new DateTime((2018 - lowAge), 5, 29); }
```

Comment:
we could make an enum that stores all of the high/lows for each range. not necessary but jus sptiballing.

Reply:
similarly, 5/29 is arbitrary here but its fine for this context